### ISSUE TITLE
Syntax Highlighting not for working for some scopes

### ISSUE THREAD

**Issue by QuantumStatic** [microsoft/pylance-release#401]
## Environment data

-   Language Server version: 2020.9.5
-   OS and version: MacOS 10.15.6
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.5

## Expected behaviour

1. Self should be orange as a parameter and when being returned .
2. magic method named call should be green should be green.
<img width="205" alt="Screenshot 2020-09-22 at 5 52 16 PM" src="https://user-images.githubusercontent.com/67118602/93881660-d264c900-fcfc-11ea-8aa7-ef47f3fd24ac.png">
3. Decorator name should be highlighted as green.
4. Function name should be green.
<img width="205" alt="Screenshot 2020-09-22 at 6 03 46 PM" src="https://user-images.githubusercontent.com/67118602/93882590-0bea0400-fcfe-11ea-8edd-a669a1c72c43.png">
5. Member Function calls are supposed to be green.
<img width="139" alt="Screenshot 2020-09-22 at 6 09 22 PM" src="https://user-images.githubusercontent.com/67118602/93883480-3ab4aa00-fcff-11ea-813e-c7aba81dddff.png">

## Actual behaviour

<img width="205" alt="Screenshot 2020-09-22 at 5 56 59 PM" src="https://user-images.githubusercontent.com/67118602/93881865-15bf3780-fcfd-11ea-878e-f439ecd212a1.png">
variable.language.special.self.python, support.function.magic.python and variable.parameter.function.python aren't overriding default colouring scheme.
<img width="205" alt="Screenshot 2020-09-22 at 6 01 41 PM" src="https://user-images.githubusercontent.com/67118602/93882624-16a49900-fcfe-11ea-8fdd-8a8ca52adbae.png">
entity.name.function.decorator.python and entity.name.function.python aren't overriding the default colour scheme.
<img width="139" alt="Screenshot 2020-09-22 at 6 11 15 PM" src="https://user-images.githubusercontent.com/67118602/93883461-34263280-fcff-11ea-8508-06b460b4ff4e.png">
meta.function-call.generic.python isn't overriding the default colour scheme.

## Logs

```
XXX
```

## Code Snippet / Additional information

```json
"scope": "meta.function-call.generic.python",
 "settings": {
              "foreground": "#6adb3e" 
             }
```

**Comment by huguesv** [comment]
Thanks for the feedback.

Semantic colorization takes precedence over the syntax colorization provided by textmate grammar. The current release is limited to the VS Code built in list of semantic token types. The known list of token and their mapping is available at https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#semantic-token-scope-map

In the next release, we've added some custom semantic token types to cover Python specific scenarios, such as the self parameter, cls parameter and dunder methods. Those will map to the existing python textmate scopes, such that colors for those identifiers will be the same with/without semantic colorization enabled.

As for decorators, we've decided to keep them colored based on their semantic type (class or function). If there's enough demand, we can add custom types for those.

For functions, we use the `function` built in token type, which VS Code maps to `entity.name.function`, so that is the color that will get picked up. We currently don't differentiate between function call/declaration, but is something we could potentially do.

FYI you can see the token/scopes via `Developer: Inspect Editor Tokens and Scopes` command. That is the easiest way to validate that tokens are correct. With just the colors, it's more difficult to tell...

**Comment by QuantumStatic** [comment]
I am trying to inform you that I use my own custom syntax colouring by customising scopes that I get from the ``` Developer: Inspect Editor Tokens and Scopes ```. I paste my code in to my settings.json file and here is the snippet of the code, I'll also be adding this to the main issue.
```
"scope": "meta.function-call.generic.python",
 "settings": {
              "foreground": "#6adb3e" 
             }
```
When I do this for the Microsoft Language Server this seems to work fine but for Pylance this doesn't work among some other scopes as mentioned above. Hope the update allows me to change for more scopes using this json snippet.

**Comment by huguesv** [comment]
Here's how you can customize colors for functions:

Using the semantic token:
```
  "editor.semanticTokenColorCustomizations": {
    "[One Dark Pro]": {
      "rules": {
        "function": "#8811ff"
      }
    }
  }
```

Or using textmate scope:
```
  "editor.tokenColorCustomizations": {
    "[One Dark Pro]": {
      "textMateRules": [
        {
          "scope": "entity.name.function",
          "settings": {
            "foreground": "#6087f3aa"
          }
        }
         ]
    }
  },
```

**Comment by huguesv** [comment]
In addition to the above changes, the next release will support the `declaration` semantic token modifier. This will allow a user/theme to customize colors differently for declarations of `function`, `property`, `variable` and `parameter`.

We've also added a `decorator` semantic token modifier, and we map `function.decorator` and `class.decorator` to the `meta.function.decorator.python` textmate scope. In One Dark Pro, this results in:
![image](https://user-images.githubusercontent.com/1696845/94483258-7207d880-018f-11eb-85a1-3c588b0f2dea.png)

If you also want the module name and variable names under a decorator to be the same color as well, then you can customize `module.decorator` and `variable.decorator` (or simply `*.decorator`) and achieve something like this:
![image](https://user-images.githubusercontent.com/1696845/94483402-a9768500-018f-11eb-8f63-bf920e87116f.png)

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.7, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202097-30-september-2020
